src/table.mjs

```diff
@@ -231,3 +271,91 @@ function likeOperand(operand) {
   return {...operand, value: `%${operand.value}%`};
 }
 
+// This function applies table cell operations to an in-memory table (array of objects).
+export function __table(source, operations) {
+  let {schema, columns} = source;
+  for (const {type, operands} of operations.filter) {
+    const column = operands.find(({type}) => type === "column").value;
+    const resolved = operands.filter(({type}) => type === "resolved");
+    switch (type) {
+      case "eq": {
+        const [{value}] = resolved;
+        source = source.filter((d) => d[column] === value);
+        break;
+      }
+      case "ne": {
+        const [{value}] = resolved;
+        source = source.filter((d) => d[column] !== value);
+        break;
+      }
+      case "c": {
+        const [{value}] = resolved;
+        source = source.filter(
+          (d) => typeof d[column] === "string" && d[column].includes(value)
+        );
+        break;
+      }
+      case "nc": {
+        const [{value}] = resolved;
+        source = source.filter(
+          (d) => typeof d[column] === "string" && !d[column].includes(value)
+        );
+        break;
+      }
+      case "in": {
+        const values = new Set(resolved.map(({value}) => value));
+        source = source.filter((d) => values.has(d[column]));
+        break;
+      }
+      case "nin": {
+        const values = new Set(resolved.map(({value}) => value));
+        source = source.filter((d) => !values.has(d[column]));
+        break;
+      }
+      case "n": {
+        source = source.filter((d) => d[column] == null);
+        break;
+      }
+      case "nn": {
+        source = source.filter((d) => d[column] != null);
+        break;
+      }
+      case "lt": {
+        const [{value}] = resolved;
+        source = source.filter((d) => d[column] < value);
+        break;
+      }
+      case "gt": {
+        const [{value}] = resolved;
+        source = source.filter((d) => d[column] > value);
+        break;
+      }
+      default:
+        throw new Error(`unknown filter type: ${type}`);
+    }
+  }
+  for (const {column, direction} of reverse(operations.sort)) {
+    const compare = direction === "desc" ? descending : ascending;
+    source.sort((a, b) => compare(a[column], b[column]));
```

Comment:
This might be mutating the input _source_. We need to make sure we make a defensive copy before sorting (but ideally we avoid unnecessary defensive copies, too).